tests/testthat/test-shinytest.R

```diff
@@ -0,0 +1,275 @@
+testthat::test_that(
+  "e2e: teal.widgets::verbatim_popup is initialized with a button that opens a modal with a verbatim text",
+  {
+    skip_if_too_deep(5)
+    module_label <- "Module with verbatim popup"
+    ui_popup_button_label <- "Open popup"
+    modal_title <- "My custom title"
+    verbatim_content_text <- "if (TRUE) { print('Popups are the best') }"
+
+    app <- TealAppDriver$new(
+      data = simple_teal_data(),
+      modules = modules(
+        module(
+          label = module_label,
+          ui = function(id) {
+            teal.widgets::verbatim_popup_ui(id, button_label = "Open popup")
+          },
+          server = eval(
+            substitute(
+              function(id) {
+                teal.widgets::verbatim_popup_srv(
+                  id,
+                  verbatim_content = verbatim_content_text,
+                  title = "modal_title",
+                  style = FALSE
+                )
+              },
+              env = list(verbatim_content_text = verbatim_content_text)
+            )
+          )
+        )
+      )
+    )
+
+    app$wait_for_idle(timeout = default_idle_timeout)
+
+    # Make sure module with a button was created.
+    testthat::expect_equal(
+      app$get_text("#teal-main_ui-root-active_tab > li.active > a"),
+      module_label
+    )
+    popup_button_element <- app$active_module_element("button")
+    testthat::expect_equal(
+      app$get_text(popup_button_element),
+      ui_popup_button_label
+    )
+
+    # Click the button.
+    app$click(selector = popup_button_element)
+    app$wait_for_idle(timeout = default_idle_timeout)
+
+    # Verify the content of the popped modal is as expected.
+    testthat::expect_equal(
+      app$get_text(".modal-title"),
+      modal_title
+    )
+
+    testthat::expect_equal(
+      app$active_module_element_text("copy_button1"),
+      "Copy to Clipboard"
+    )
+    testthat::expect_equal(
+      app$get_text("#shiny-modal > div > div > div.modal-body > div > button:nth-child(2)"),
+      "Dismiss"
+    )
+    testthat::expect_equal(
+      app$active_module_element_text("verbatim_content"),
+      verbatim_content_text
+    )
+
+    # Modal is closed, once the button is clicked.
+    app$click(selector = "#shiny-modal-wrapper button[data-dismiss='modal']")
+    # There are two Dismiss buttons.
+    app$wait_for_idle(timeout = default_idle_timeout)
+    testthat::expect_null(app$get_html("#shiny-modal-wrapper"))
+
+    app$stop()
+  }
+)
+
+testthat::test_that(
```

Comment:
Please split the content into multiple files with naming convention identical to teal (`test-shinytest2-<name of a function of a file within R directory>.R`

Reply:
totally :) just though they same about it